### ISSUE TITLE
Java 16 support - Research Spike

### ISSUE THREAD

**Issue by jodeev** [newrelic/newrelic-java-agent#215]

**Comment by mbazos** [comment]
Hi @jodeev was going to file an issue as we are using the latest agent 6.5.0 and noticed this error when using Java 16:
```
Java version is: 16. This version of the New Relic Agent does not support versions of Java greater than 15.
```

Glad to see it's being worked on.

**Comment by mattnelson** [comment]
Are the changes to support running on 16 or supporting language features added in 16?

The PR for 15 appears to have been isolated to just running https://github.com/newrelic/newrelic-java-agent/pull/56

**Comment by ryanb93** [comment]
Is there a rough/estimated timeline for supporting Java 16? 

Would like to upgrade our stack to take advantage of new language features but NewRelic support would be a blocker for our organisation.

**Comment by Stephan202** [comment]
Could the scope of this task be extended to Java 17? It's [currently](https://openjdk.java.net/projects/jdk/17/) in "rampdown phase one" and is scheduled for release on September 14, i.e. less than two months from now. As New Relic customers we'd like to adopt this new LTS release as soon as it comes out.

**Comment by Stephan202** [comment]
JDK 17's "initial release candidate" was released last week. The "final release candidate" is scheduled for next week. The GA release is scheduled for exactly one month from now. Can somebody from New Relic indicate when this LTS release is expected to be officially supported by the Java Agent? We have a whole team eager to adopt Java 17. I filed #370 to take care of some of the grunt work involved in this effort.
